pallets/offchain-worker/src/lib.rs

```diff
@@ -294,50 +479,124 @@ impl<T: Trait> Module<T> {
+		//  "15EW3AMRm2yP6LEF5YKKLYwvphy3DmMqN6":{"final_balance":0,"n_tx":4,"total_received":310925609}
+	  // }
 		let val = lite_json::parse_json(price_str);
-		let balance = val.ok().and_then(|v| match v {
+		let mut balance_vec: Vec<u128> = Vec::new();
+
+		val.ok().and_then(|v| match v {
 			JsonValue::Object(obj) => {
-				obj.into_iter()
-					.find(|(k, _)| {
-						let mut chars = "result".chars();
-						k.iter().all(|k| Some(*k) == chars.next())
-					})
-					.and_then(|v| match v.1 {
-						JsonValue::String(balance) => Some(balance),
-						_ => None,
-					})
+				for each in obj {
+					match each.1 {
+						JsonValue::Object(balance_pairs) => {
+							balance_vec.push(balance_pairs.into_iter().find(|(k, _)|{
+								let mut matching_chars = "final_balance".chars();
+								k.iter().all(|k| Some(*k) == matching_chars.next())
+							})
+							.and_then(|v| match v.1 {
+								JsonValue::Number(balance) => {
+									if balance.fraction != 0 || balance.fraction_length != 0 || balance.integer < 0 {
+										// Number received with fraction or negative, abort this request
+										None
+									} else {
+										Some( 
+											if balance.exponent == 0 {
+												balance.integer as u128 
+											} else {
+												balance.integer as u128 * 10u128.pow(balance.exponent as u32)
+											})
+									}
+								},
+								_ => None,
+							})?);
+						},
+						_ => ()
+					}
+				};
+				Some(balance_vec)
 			},
 			_ => None
-		})?;
-		Some(balance)
+		})
 	}
 
-	// U64 number string to u64
-	pub fn chars_to_u64(vec: Vec<char>) -> Result<u64, &'static str> {
-		let mut result: u64 = 0;
-		for item in vec {
-			let n = item.to_digit(10);
+	// Parse the balance from infura response
+	fn parse_infura_balances(price_str: &str) -> Option<Vec<u128>> {
+		//[
+		//  {"jsonrpc":"2.0","id":1,"result":"0x4563918244f40000"},
+		//  {"jsonrpc":"2.0","id":1,"result":"0xff"}
+		//]
+		let val = lite_json::parse_json(price_str);
+		let mut balance_vec: Vec<u128> = Vec::new();
+
+		val.ok().and_then(|v| { 
+				match v {
+					JsonValue::Array(res_array) => {
+						for each in res_array {
+							match each {
+								JsonValue::Object(obj) => {
+									balance_vec.push({
+										obj.into_iter().find(|(k, _)|{
+											let mut chars = "result".chars(); 
+											k.iter().all(|k| Some(*k) == chars.next())
+										}).and_then(|v|{
+											match v.1 {
+												JsonValue::String(balance) => {
+													match Self::chars_to_u128(balance){
+														Ok(b) => Some(b),
+														Err(_) => None,
+													}
+												},
+												_ => None,
+											}
+										})?
+									});
+								},
+								_ => (),
+							}
+						};
+						Some(balance_vec)
+					},
+					_ => None,
+				}
+			})
+	}
+
+	// u128 number string to u128
+	pub fn chars_to_u128(vec: Vec<char>) -> Result<u128, &'static str> {
```

Comment:
When the code gets more and more, you may want to move these helper functions to a separate file and get their tests also there to make the code clearer